source/simulator/helper_functions.cc

```diff
@@ -1033,6 +1033,7 @@ namespace aspect
                         (theta, abs((min_solution_exact_global-local_solution_average)
                                     /(min_solution_local-local_solution_average)));
                 //Modify the advection degrees of freedom of the numerical solution
+                IndexSet system_index_set = dof_handler.locally_owned_dofs();
```

Comment:
I added a IndexSet here although I am not sure if it's the correct way to do it. However, the results with multiple processors changes a lot  after adding this and making a ifstatement later to test if the dof of the current local cell is also a dof of current processor.